### PR TITLE
Display single movie using api

### PR DESCRIPTION
**Pull request by priscillaapaxton**
# Description
Single movie is now displayed using relevant API. 

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
# How Has This Been Tested?
n/a
# Checklist:
- [ ] My code follows the TDD (Test Driven Development) style guidelines of this project
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have written modular, reusable code that follows SRP (Single Responsibility Principle)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new problems
- [x] I have made corresponding changes to the project board
# Notes & Next:
Single movie class could be refactored to make the set state/state not so lengthy, but I am not sure how to do that yet. Loading screen and lighthouse test for accessibility still need to be done

### CODE REVIEW

**Comment by mattsaqueli** [comment]
Awesome thank you! Next we can start working on the accessibility test and loading screen.

File: src/App.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ class App extends React.Component {
   }
 
   componentDidMount = () => {
-    apiCalls.getData().then((data) => {
+    apiCalls.getAllMovies().then((data) => {
       this.setState({
         allMovies : data.movies,
         SelectedMovie: null
```
Comment (niksseif) [inline src/App.js:22]: Assigning 'null' to 'SelectedMovie' might create some bugs in your code. The best practice is always to assign the data type that you're expecting to receive from the API. Is 'SelectedMovie' an object, array, number, or string?

File: src/SingleMovie.js [hunk 2]
```diff
@@ -1,18 +1,57 @@
...
-        />
-        <button onClick={(e) => {
-          e.preventDefault()
-          props.returnHome()
-        }}>HOME</button>
+class SingleMovie extends React.Component {
+  constructor() {
+    super()
+    this.state= {
+      currentMoviePoster: null,
```
Comment (niksseif) [inline src/SingleMovie.js:9]: Do you need to have all of these as a state of your component? How can you DRY this state up?

File: src/App.js [hunk 3]
```diff
@@ -26,9 +26,10 @@ class App extends React.Component {
 
   displaySelectedMovie = (id) => {
     const filteredMovie = this.state.allMovies.filter(movie => movie.id === id)
+    // console.log('line28', filteredMovie[0].id)
     this.setState({
       test: false,
-      SelectedMovie: filteredMovie
+      SelectedMovie: filteredMovie[0].id
```
Comment (niksseif) [inline src/App.js:32]: What is the length of this 'filteredMovie' array? Aren't the IDs unique, and shouldn't we only return one movie that matches the ID? It seems a bit hardcoded to me, using the index number here to get that specific movie from 'filteredMovie'.

File: src/apiCalls.js [hunk 4]
```diff
@@ -13,4 +13,19 @@ export function getData () {
...
+export function getSingleMovie (movieId) {
+  return fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${movieId}`)
+  .then((response) => {
+    if (!response.ok) {
+      throw new Error(`${response.status}`)
+    } else {
+      return response.json()
+    }
+  })
+  .catch((err) => {
```
Comment (niksseif) [inline src/apiCalls.js:25]: Nice job using throw error - There's no need to have the catch block here. You can move the catch block to where you're calling this function within the componentDidMount() method.

**Review by niksseif — COMMENTED** [review]
Hey team! I've left a few suggestions for you to consider. Please let me know if you have any questions.

**Pull request merged**